### DISCUSSION TITLE
Promptless Logout

### DISCUSSION THREAD

**Discussion by kwaksaewon** [mstaal/msal_streamlit_authentication#7] [Ideas]
As of right now, there is logout_request parameter in msal_authentication.

To do a promptless logout, according to the [promptless logout documentation](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/logout.md#promptless-logout), we need to extract or set the logoutHint and pass that into [msalInstance.logoutPopup](https://github.com/mstaal/msal_streamlit_authentication/blob/930f982dfe67473ce5b44d3bc2aa2fb1709a5d3b/msal_streamlit_authentication/frontend/src/Authentication.tsx#L52).
Could promptless logout be supported?

**Comment by mstaal** [comment]
That’s an interesting suggestion. I will try to look into it soon. I
apologize that I may be slow at allocating the relevant time!

tor. 8. jun. 2023 kl. 23.42 skrev kwaksaewon ***@***.***>:

> As of right now, there is logout_request parameter in msal_authentication.
>
> To do a promptless logout, according to the promptless logout
> documentation
> <https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/logout.md#promptless-logout>,
> we need to extract or set the logoutHint and pass that into
> msalInstance.logoutPopup
> <https://github.com/mstaal/msal_streamlit_authentication/blob/930f982dfe67473ce5b44d3bc2aa2fb1709a5d3b/msal_streamlit_authentication/frontend/src/Authentication.tsx#L52>
> .
> Could promptless logout be supported?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mstaal/msal_streamlit_authentication/discussions/7>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4LJ36NJF7RUNF2PPADWXLXKJBMBANCNFSM6AAAAAAY7223PU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Venligst,
Michael

**Reply by kwaksaewon** [reply]
Thanks!
